vgen/src/test/java/io/gapi/vgen/testdata/go_library.baseline

```diff
@@ -191,14 +189,14 @@ func (book Book) Path() (string, error) {
 
 // CreateShelf creates a shelf, and returns the new Shelf.
 func (client *Client) CreateShelf(ctx context.Context, req *google_example_library_v1.CreateShelfRequest, opts ...gax.CallOption) (*google_example_library_v1.Shelf, error) {
-	callOpts := append(nonIdempotentRetryCodes(), defaultRetryParams(), client.CallOptions["CreateShelf"]...)
-	callOpts = append(callOpts, opts...)
+	defaultCallOpts := append(nonIdempotentRetryCodes(), defaultRetryParams()...)
+	callOpts := append(client.callOptions["CreateShelf"], opts...)
```

Comment:
Yes, this is buggy. append does try to modify its first argument if there is space.

I suggest simple, linear code:

var callOpts []gax.CallOption
callOpts = append(callOpts, thing1)
callOpts = append(callOpts, thing2)
...

(This is not optimal in space or time, but it almost certainly won't matter.)